src/client/opamArg.ml

```diff
@@ -782,28 +774,263 @@ let opamlist_columns =
+            when cond_deprecated cli removal ->
+            ok_deprecated_flag cli since removal instead flags;
+            neww,rm
+          | _ -> neww,rm) ([],[]) selected
+    in
+    match neww, rm with
+    | [flags, c], [] ->
+      error_new_flag cli c flags
+    | [], [flags, (c, instead)] ->
+      error_rm_flag cli c instead flags
+    | [], [] -> `Ok selected
+    | _::_, []->
+      let options, clis = List.split neww in
+      let msg =
+        Printf.sprintf
+          "%s are only available since respectively versions %s of the opam \
+           CLI, but version %s has been requested."
+          (OpamStd.Format.pretty_list (List.map get_long_form options) )
+          (OpamStd.Format.pretty_list
+             (List.map OpamCLIVersion.to_string clis))
+          (OpamCLIVersion.to_string cli)
+      in
+      `Error (false,msg)
+    | [], _::_->
+      let options, clis = List.split rm in
+      let clis = List.split clis |> fst in
+      let msg =
+        Printf.sprintf
+          "%s were removed in, respectively, %s versions of the opam CLI, \
+           but version %s has been requested."
+          (OpamStd.Format.pretty_list (List.map get_long_form options) )
+          (OpamStd.Format.pretty_list
+             (List.map OpamCLIVersion.to_string clis))
+          (OpamCLIVersion.to_string cli)
+      in
+      `Error (false,msg)
+    | _,_ ->
+      let news, nclis = List.split neww in
+      let rms, rclis = List.split rm in
+      let rclis = List.split rclis |> fst in
+      let msg =
+        Printf.sprintf
+          "%s are only available since respectively versions %s of the opam \
+           CLI,and %s were removed in respectively versions %s, \
+           but version %s has been requested."
+          (OpamStd.Format.pretty_list (List.map get_long_form news))
+          (OpamStd.Format.pretty_list
+             (List.map OpamCLIVersion.to_string nclis))
+          (OpamStd.Format.pretty_list (List.map get_long_form rms))
+          (OpamStd.Format.pretty_list
+             (List.map OpamCLIVersion.to_string rclis))
+          (OpamCLIVersion.to_string cli)
+      in
+      `Error (false,msg)
+  in
+  term_cli_check ~check Arg.(vflag_all default info_flags)
 
-let mk_tristate_opt ?section flags value doc =
+let mk_tristate_opt ~cli validity ?section flags value doc =
+  let doc = update_doc_w_cli doc ~cli validity in
   let doc = Arg.info ?docs:section ~docv:value ~doc flags in
-  Arg.(value & opt (some (enum when_enum)) None & doc)
+  let check elem = check_cli_validity cli validity elem flags in
+  term_cli_check ~check Arg.(opt (some (enum when_enum)) None & doc)
+
+(*
+let mk_pos ?rev p content default ~docv ~doc =
+Arg.(value & pos ?rev p content default & info ~docv ~doc [])
+
+let mk_pos ~cli ~validity ?rev p content default ~docv ~doc =
+  let doc = update_doc_w_cli doc ~cli validity in
+  Arg.(value & pos ?rev p content default & info ~docv ~doc [])
+*)
```

Comment:
I didn't neither treated positional arguments, as they are not prone to change.

Reply:
Agreed!